common/src/main/java/me/lucko/luckperms/common/calculator/LuckyPerms.java

```diff
@@ -0,0 +1,145 @@
+ *  SOFTWARE.
+ */
+
+package me.lucko.luckperms.common.calculator;
+
+import net.luckperms.api.util.Tristate;
+
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * The LuckPerms Luckiness Calculator (tm)
+ */
+public final class LuckyPerms {
+
+    /** A sequence of the first 1000 lucky numbers. */
+    private static final Set<Integer> LUCKY_NUMBERS = generateLuckyNumbers(1000);
+    /** Is it an unlucky day? */
+    private static final boolean UNLUCKY_DAY;
+    /** The unlucky number. */
+    private static final int UNLUCKY_NUMBER = 666;
+    /** The lucky number */
+    private static final int LUCKY_NUMBER;
+
+    static {
+        LocalDate date = LocalDate.now();
+        UNLUCKY_DAY = date.getDayOfWeek() == DayOfWeek.FRIDAY && date.getDayOfMonth() == 13;
+
+        if (Locale.getDefault().getLanguage().equals(Locale.CHINESE.getLanguage())) {
+            // https://en.wikipedia.org/wiki/Chinese_numerology#Eight
+            LUCKY_NUMBER = 8;
+        } else {
+            // https://en.wikipedia.org/wiki/7
+            LUCKY_NUMBER = 7;
+        }
+    }
+
+    /** A ~magical~ counter used when determining luckiness. */
+    private final AtomicInteger counter;
+
+    public LuckyPerms() {
+        this.counter = new AtomicInteger();
+    }
+
+    /**
+     * Uses highly advanced, patented LuckPerms(tm) calculations to determine
+     * the current luckiness measure.
+     *
+     * <p>Returns {@link Tristate#TRUE} if lucky, {@link Tristate#FALSE} if unlucky,
+     * and {@link Tristate#UNDEFINED} if unable to determine.</p>
+     *
+     * @return the calculated luckiness measure
+     */
+    public Tristate calculateLuckiness() {
+        int i = this.counter.incrementAndGet();
+
+        // is the counter divisible by the unlucky number? oh no.. that's not a good sign!
+        if (i % UNLUCKY_NUMBER == 0) {
+            return Tristate.FALSE;
+        }
+
+        // got lucky on a dice roll & the counter is in the lucky numbers set? must be lucky!
+        if (isEven(rollDice()) && LUCKY_NUMBERS.contains(i % 1000)) {
+            return Tristate.TRUE;
+        }
+
+        // is it Friday 13th? I sure hope not.
+        if (UNLUCKY_DAY) {
+            return Tristate.FALSE;
+        }
+
+        // is the current system time divisible by the lucky number? sounds promising?!
+        if ((System.currentTimeMillis() / 1000) % LUCKY_NUMBER == 0) {
+            return Tristate.TRUE;
+        }
+
+        // well, this is embarrassing.
+        return Tristate.UNDEFINED;
+    }
+
+    // Roll a 6-sided fair dice.
+    private static int rollDice() {
+        return ThreadLocalRandom.current().nextInt(6) + 1;
+    }
+
+    // Is the number even?
+    private static boolean isEven(int i) {
```

Comment:
This seems like un-neccesary code, is there maybe a library that could be imported to serve this function? 
I know in JavaScript you'd just use https://www.npmjs.com/package/is-even - are JavaScript modules compatible with Java due to the similar name?

Reply:
Salutations @codingJWilliams

As the magnanimous director of scrutiny here at the LuckPerms conglomerate, I'd have to hazard a guess that yes there is a very marked similarity in the naming schemes of both code-ery-thingies. 

As such, it is indeed almost certain that this can be handled by the same JavaScript module you've just mentioned.

As such, the author of this code will be scrutinized very closely, while also being notified of this close scrutiny.

As such, this close scrutiny will likely effect a change in the disposition of the author of this code, this code that currently fails to take advantage of such useful packages as you have just described.

As such, we can hopefully have this resolved as soon as possible. 

As such, I'd like to thank you for your initiation of this conversation, and would like to let you know that you also have been scrutinized and not found lacking.

As such, you are in excellent condition to continue as the Senior Lead Managing Director of LuckPerms' taco department.

As such, I'd like to wish you a very nice afternoon.

With gratitude,

Larry
Magnanimous Director of Scrutiny with the LuckPerms Conglomerate.